src/main/java/io/xpire/model/item/Quantity.java

```diff
@@ -56,13 +57,16 @@ private Quantity(int quantity) throws ParseException {
     }
 
     public static boolean isValidQuantity(String test) {
-        return StringUtil.isNonNegativeInteger(test);
+        return StringUtil.isNonNegativeInteger(test) && Integer.parseInt(test) <= MAX_VALUE;
```

Comment:
I actually added an isExceedingLimit method to the StringUtil class so you can use that

Reply:
noted!